### DISCUSSION TITLE
Conventions for multiple inheritance

### DISCUSSION THREAD

**Discussion by mjclarke94** [pydantic/pydantic#5974] [Question] [answered]
Consider the following system:

```
from pydantic import BaseModel
from abc import abstractproperty
from datetime import timedelta

class Base(BaseModel):
    identifier: int

    @abstractproperty
    def description(self):
        pass

class HasDuration(BaseModel):
    duration: timedelta
    
class ModelA(HasDuration, Base):
    @property
    def description(self):
        return f'{self.duration}'

a = ModelA(duration=timedelta(seconds = 3) , identifier=1)

assert hasattr(a, 'duration')
assert hasattr(a, 'identifier')
assert hasattr(a, 'description')
assert ModelA.parse_raw(a.json()) == a
```

As far as I can tell, this works as expected, but I've not seen any examples of people constructing models using multiple inheritance. Is there some issue with taking this approach?

**Comment by dmontagu** [comment, marked as answer]
I would generally advise against this pattern when possible because, in my experience, it just makes it a bit harder to navigate the codebase. But I think it's something we have _purposely_ tried to support if helpful for any reason (e.g., for the use of mixin classes).

If it works for you I wouldn't consider it a misuse of the library or anything.

**Reply by mjclarke94** [reply]
Cool, thanks :) Generally I'm adverse to the mixin pattern but we have a use case where the alternative would be horrendous to maintain. Fortunately, it's for an internal library so the suffering will be limited to a few people!
